### PR TITLE
chore(typings): add MockOption["customEgg"]: boolean for framework usage

### PR DESCRIPTION
**Pull request by waitingsong**
variable from format_options.js

##### Checklist

- [ ] `npm test` passes
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message follows commit guidelines

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -87,6 +87,11 @@ export interface MockOption {
    * Enable typescript support
    */
   typescript?: boolean;
+
+  /**
+   * framework support
+   */
+  customEgg?: boolean;
```
Comment (atian25) [inline index.d.ts]: `customEgg` is deprecated, use `framework` instead.

Comment (waitingsong) [reply index.d.ts]: 原来如此。我照着 `egg-example/framework/../test/lib/framework.test.js` 写的

```js
describe('test/lib/framework.test.js', () => {
  let app;
  before(() => {
    app = mm.app({
      baseDir: path.join(__dirname, '../../../app'),
      customEgg: true,
    });
    return app.ready();
  });
```

Comment (waitingsong) [reply index.d.ts]: 不过 framework 类型是 string,  我就传 options.framework 值？

Comment (atian25) [reply index.d.ts]: framework 应该是支持 string|boolean 的吧，你看下用例

Comment (waitingsong) [reply index.d.ts]: 我这样取值，测试通过了
`      framework: path.join(__dirname, '../../'), `

Comment (atian25) [reply index.d.ts]: 应该是支持 boolean 的，可以提个 PR

**Pull request closed without merging**